clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -507,6 +507,7 @@
     private final Metadata metadata;
     private final long retryBackoffMs;
     private final long requestTimeoutMs;
+    private final boolean autoCommitEnabled;
```

Comment:
Yes, that's right. The flag needs to be passed to the coordinator, which should work without the member variable. Updating the patch now ... Thanks.